tests/queries/0_stateless/03545_union_allow_column_with_no_common_type.sql

```diff
@@ -0,0 +1,8 @@
+SELECT 1 AS x UNION ALL SELECT 'Hello' ORDER BY 'x' SETTINGS use_variant_as_common_type = 1;
```

Comment:
This `ORDER BY` doesn't actually order the output: https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=83246&sha=64ddd176118afb817efb66aaee41aaa0ac09c423&name_0=PR&name_1=Stateless+tests+%28amd_asan%2C+flaky+check%29&name_2=Tests